### PR TITLE
Enable SonarCloud scan

### PR DESCRIPTION
**Pull request by gsmet**
New attempt based on https://github.com/quarkusio/quarkus/pull/13444 .

### CODE REVIEW

File: .github/workflows/sonarcloud.yml [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+      - name: Cache Maven packages
+        uses: actions/cache@v2
+        with:
+          path: ~/.m2
+          key: ${{ runner.os }}-m2-${{ steps.get-date.outputs.date }}
+      - name: Build and analyze
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+          SONAR_TOKEN: ${{ secrets.SONAR_TOKEN }}
+        run: ./mvnw -B -Dquickly install org.sonarsource.scanner.maven:sonar-maven-plugin:sonar -Dsonar.organization=quarkusio -Dsonar.projectKey=quarkusio_quarkus -Dsonar.host.url=https://sonarcloud.io -Dsonar.projectName=Quarkus -Dsonar.java.libraries=/home/runner/.m2/repository/**/*.jar
```
Comment (famod) [inline .github/workflows/sonarcloud.yml:39]: Better use `quickly-ci`, no?

Comment (famod) [reply .github/workflows/sonarcloud.yml:39]:
```suggestion
        run: ./mvnw -e -B --settings .github/mvn-settings.xml -Dquickly-ci install org.sonarsource.scanner.maven:sonar-maven-plugin:sonar -Dsonar.organization=quarkusio -Dsonar.projectKey=quarkusio_quarkus -Dsonar.host.url=https://sonarcloud.io -Dsonar.projectName=Quarkus -Dsonar.java.libraries=/home/runner/.m2/repository/**/*.jar
```

**Comment by famod** [comment]
Nice! Are you planning to add that for PRs as well? Probably too much load, right?

**Comment by gsmet** [comment]
@famod yes: https://github.com/quarkusio/quarkus/pull/16178/files#diff-5895707186d21bd3672c2faf7743c1cd2d656de088cdec80dc676ed8e3f22bceR6

From what I know, SonarCloud is supposed to be able to only report issues on new code. I have no idea how it will work in practice though.

Let's get it working and then we can discuss :).

I made an adjustment to only enable manual execution. Let's see how it goes first.

**Comment by gsmet** [comment]
Let's merge it and iterate from there.

**Pull request merged**